src/app/system/manage-surveys/create-survey/create-survey.component.html

```diff
@@ -0,0 +1,163 @@
+<div class="container">
+
+  <form [formGroup]="surveyForm" (ngSubmit)="submit()">
+
+    <div fxLayout="column" fxLayoutGap="4%">
+
+      <mat-card>
+
+        <mat-card-content>
+
+          <div fxLayout="row wrap" fxLayoutGap="2%" fxLayout.lt-md="column">
+
+            <mat-form-field fxFlex="48%">
+              <mat-label>Key</mat-label>
+              <input matInput maxlength="32" required formControlName="key">
+              <mat-error *ngIf="surveyForm.controls.key.hasError('required')">
+                Key is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+            
+            <mat-form-field fxFlex="48%">
+              <mat-label>Name</mat-label>
+              <input matInput maxlength="255" required formControlName="name">
+              <mat-error *ngIf="surveyForm.controls.name.hasError('required')">
+                Name is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="48%">
+              <mat-label>Country Code</mat-label>
+              <input matInput maxlength="2" required formControlName="countryCode">
+              <mat-error *ngIf="surveyForm.controls.countryCode.hasError('required')">
+                Country Code is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="surveyForm.controls.countryCode.hasError('pattern')">
+                Country Code <strong>must consist of 2 uppercase characters</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="98%">
+              <mat-label>Description</mat-label>
+              <textarea matInput formControlName="description"></textarea>
+            </mat-form-field>
+
+            <div fxFlexFill fxLayout="row" fxLayoutAlign="center">
+              <button type="button" mat-raised-button color="primary" *ngIf="questionDatas.length !== 0" [disabled]="!(this.surveyForm.get('key').valid && this.surveyForm.get('name').valid && this.surveyForm.get('countryCode').valid)" (click)="addQuestion()">
```

Comment:
Clicking the button adds the first question, and adding the first question allows the submit and add question button to appear through *ngFor. This prevents the need to create a new function to 'set' the questions up.